crux-core/src/crux/object_store.clj

```diff
@@ -19,20 +19,26 @@
 (defrecord KvObjectStore [kv]
   db/ObjectStore
   (get-single-object [this snapshot k]
-    (let [doc-key (c/->id-buffer k)
-          seek-k (c/encode-doc-key-to (.get i/seek-buffer-tl) doc-key)]
-      (some->> (kv/get-value snapshot seek-k)
-               (DirectBufferInputStream.)
-               (DataInputStream.)
-               (nippy/thaw-from-in!))))
+    (if (satisfies? kv/KvSnapshot snapshot)
+      (let [doc-key (c/->id-buffer k)
+            seek-k (c/encode-doc-key-to (.get i/seek-buffer-tl) doc-key)]
+        (some->> (kv/get-value snapshot seek-k)
+                 (DirectBufferInputStream.)
+                 (DataInputStream.)
+                 (nippy/thaw-from-in!)))
+      (with-open [snapshot (kv/new-snapshot kv)]
+        (db/get-single-object this snapshot k))))
 
   (get-objects [this snapshot ks]
-    (->> (for [k ks
-               :let [seek-k (c/encode-doc-key-to (.get i/seek-buffer-tl) (c/->id-buffer k))
-                     doc (some-> (kv/get-value snapshot seek-k) <-nippy-buffer)]
-               :when doc]
-           [k doc])
-         (into {})))
+    (if (satisfies? kv/KvSnapshot snapshot)
```

Comment:
satisfies is proved to be slow..

Reply:
As discussed, this is a known issue for now, if we run into noticeable performance issues we'll solve it. Eventually the need for this should go away if for example the object store goes and gets deprecated by the document store. Currently there's only one indexer - this check is there if you would use a new indexer together with the KvObjectStore.